src/test/java/com/snowflake/kafka/connector/records/RecordContentTest.java

```diff
@@ -112,4 +118,119 @@ public void test() throws IOException
     content = new SnowflakeRecordContent(null, jsonMap);
     assert content.getData()[0].toString().equals("{\"int8\":12,\"int16\":12,\"int32\":12,\"int64\":12,\"float32\":12.2,\"float64\":12.2,\"boolean\":true,\"string\":\"foo\",\"bytes\":\"Zm9v\",\"array\":[\"a\",\"b\",\"c\"],\"map\":{\"field\":1},\"mapNonStringKeys\":[[1,1]]}");
   }
+
+  @Test(expected = SnowflakeKafkaConnectorException.class)
+  public void testEmptyValue()
+  {
+    RecordService service = new RecordService();
+
+    SinkRecord record = new SinkRecord(
+      topic, partition,
+      null, null,
+      Schema.STRING_SCHEMA, null,
+      partition
+    );
+    service.processRecord(record);
+  }
+
+  @Test(expected = SnowflakeKafkaConnectorException.class)
+  public void testEmptyValueSchema() throws IOException
+  {
+    JsonNode data = mapper.readTree("{\"name\":123}");
+    SnowflakeRecordContent content = new SnowflakeRecordContent(data);
+    RecordService service = new RecordService();
+
+    SinkRecord record = new SinkRecord(
+      topic, partition,
+      null, null,
+      null, content,
+      partition
+    );
+    service.processRecord(record);
+  }
+
+  @Test(expected = SnowflakeKafkaConnectorException.class)
+  public void testWrongValueSchema() throws IOException
+  {
+    JsonNode data = mapper.readTree("{\"name\":123}");
+    SnowflakeRecordContent content = new SnowflakeRecordContent(data);
+    RecordService service = new RecordService();
+
+    SinkRecord record = new SinkRecord(
+      topic, partition,
+      null, null,
+      SchemaBuilder.string().name("aName").build(), content,
+      partition
+    );
+    // TODO: Fix this after stability push, if schema does not have a name
```

Comment:
file Jira for this todo, and write the JIRA # in the todo

Reply:
ack